controller/backup_controller.go

```diff
@@ -284,15 +283,19 @@ func (bc *BackupController) reconcile(backupName string) (err error) {
 
 	// Perform backup snapshot to remote backup target
 	if backup.Status.LastSyncedAt.IsZero() && backup.Spec.SnapshotName != "" {
-		volume, err := bc.ds.GetVolume(backupVolumeName)
-		if err != nil {
-			return err
-		}
-		skip, err := bc.backupCreation(backupTarget, volume, backup)
-		if err != nil {
+		if err := bc.backupCreation(backup, backupVolumeName, backupTarget); err != nil {
+			log.WithError(err).Error("Failed to sync engine snapshot backup")
 			return err
 		}
-		if skip {
+
+		switch backup.Status.State {
+		case longhorn.BackupStateNew, longhorn.BackupStateInProgress:
+			return nil
+		case longhorn.BackupStateCompleted:
+			bc.disableBackupMonitor(backup.Name)
+		case longhorn.BackupStateError, longhorn.BackupStateUnknown:
+			backup.Status.LastSyncedAt = syncTime
+			bc.disableBackupMonitor(backup.Name)
 			return nil
 		}
 	}
```

Comment:
Can we refactor it like the following? The method namings here are not important...
```
syncWithReplica := backup.Status.LastSyncedAt.IsZero() && backup.Spec.SnapshotName != ""
if syncWithReplica {
    checkMonitor(backup)  // enable or disable monitor based on backup current state
    syncWIthMonitor()
    WrapUp()  // Disable monitor and update backup volume
    return
}

// The backup creation is complete, then the source of truth becomes the remote backup target
```

Reply:
Yes, we could consider it once we have a consensus on [2].

Reply:
Updated.